File: 43. Multiply Strings.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        int n = num1.size() + num2.size();
        vector<int> c(n, 0);
        for (int i = num1.size() - 1; i >= 0 ; i--)
        {
            for (int j = num2.size() - 1; j >= 0; j--)
            {
                c[i+j+1] += (num1[i] - '0')*(num2[j] - '0');
            }
        }
        
        for (int i = n-1; i > 0; i--)
        {
            c[i-1] += c[i]/10;
            c[i] %= 10;
        }
        
        int i = 0;
        while (c[i] == 0)
            i++;
        
        string res;
        while (i < n)
        {
            res.push_back(c[i++] + '0');
        }
        if (res.empty()) return "0";
        return res;
    }
};

File: python/43. Multiply Strings.py
class Solution(object):
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        n = len(num1) + len(num2)
        c = [0]*n
        for i in range(len(num1)-1, -1, -1):
            for j in range(len(num2)-1, -1, -1):
                c[i+j+1] += (ord(num1[i]) - ord('0')) * (ord(num2[j]) - ord('0'))

        for i in range(n-1, 0, -1):
            c[i-1] += c[i]/10
            c[i] %= 10

        i = 0
        while i<n and c[i]==0:
            i += 1
        if i==n: return "0"
        return ''.join([str(i) for i in c[i:]])

if __name__ == '__main__':
    su = Solution()
    print su.multiply("0", "0")        